x/auth/keeper/keeper.go

```diff
@@ -275,3 +278,43 @@ func (ak AccountKeeper) GetParams(ctx context.Context) (params types.Params) {
 	}
 	return params
 }
+
+func (ak AccountKeeper) NonAtomicMsgsExec(ctx context.Context, signer sdk.AccAddress, msgs []sdk.Msg) ([]*types.NonAtomicExecResult, error) {
+	var msgsAny []*codectypes.Any
+	msgResponses := make([]*types.NonAtomicExecResult, 0, len(msgs))
+
+	for _, msg := range msgs {
+		if m, ok := msg.(sdk.HasValidateBasic); ok {
+			if err := m.ValidateBasic(); err != nil {
+				value := &types.NonAtomicExecResult{Error: err.Error()}
+				msgResponses = append(msgResponses, value)
+				continue
+			}
+		}
+
+		anyValue, err := codectypes.NewAnyWithValue(msg)
+		if err != nil {
+			value := &types.NonAtomicExecResult{Error: err.Error()}
+			msgResponses = append(msgResponses, value)
+			continue
+		}
+		msgsAny = append(msgsAny, anyValue)
+
+	}
+
+	results, err := ak.AccountsModKeeper.SendAnyMessages(ctx, signer, msgsAny)
```

Comment:
seems a bit overkill to depend on accounts to use this function, can it be avoided?

Reply:
Initially, I used baseapp's msg router to handle routing. But that'd require passing baseapp's router to auth keeper and eventually leading to baseapp usage everywhere auth keeper is needed. After frojdi's review and I had a discussion with frojdi, he suggested having accounts x/keeper interface and using this method directly. Because we could directly send  `Any` without even  unpacking them and it will check the signer, which removes a lot of boilerplate, and avoiding usage of baseapp everywhere else.